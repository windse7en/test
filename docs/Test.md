# Terra Test

 {insert description}

## Getting Started

- Install with [npmjs](https://www.npmjs.com): 
  - `npm install terra-test` 
  - `yarn install terra-test` 
- [Download the latest version](https://github.com/cerner/terra-test/archive/master.zip)
- Clone the repo: `git clone https://github.com/cerner/terra-test.git`
