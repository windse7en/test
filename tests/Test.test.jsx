import React from 'react';
import initStoryshots from 'storyshots';
import Test from '../src/Test';

const defaultVariety = <Test />;
const primaryVariety = <Test name="primary" variant="terra-Test--primary" />;

// Run snapshot tests for react-storybook
initStoryshots();

// Snapshot Tests
it('should render a default component', () => {
  const wrapper = shallow(defaultVariety);
  expect(wrapper).toMatchSnapshot();
});

it('should render a primary component', () => {
  const wrapper = shallow(primaryVariety);
  expect(wrapper).toMatchSnapshot();
});


// Prop Tests
it('should have the class terra-Test--default', () => {
  const wrapper = shallow(defaultVariety);
  expect(wrapper.prop('className')).toContain('terra-Test terra-Test--default');
});

it('should have the class terra-Test--primary', () => {
  const wrapper = shallow(primaryVariety);
  expect(wrapper.prop('className')).toContain('terra-Test terra-Test--primary');
});


// Event Tests
it('should toggle the class u-selected on default', () => {
  const wrapper = shallow(defaultVariety);
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-Test--default').simulate('click');
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-Test--default').simulate('click');
  expect(wrapper).toMatchSnapshot();
});

it('should toggle the class u-selected on primary', () => {
  const wrapper = shallow(primaryVariety);
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-Test--primary').simulate('click');
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-Test--primary').simulate('click');
  expect(wrapper).toMatchSnapshot();
});
