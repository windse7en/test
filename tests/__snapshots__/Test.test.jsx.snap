exports[`Storyshots Test  Without props 1`] = `
<span>
  <button
    className="terra-Test terra-Test--default"
    onClick={[Function]}>
    Terra, 
    default
  </button>
</span>
`;

exports[`test should render a default component 1`] = `
<button
  className="terra-Test terra-Test--default"
  onClick={[Function]}>
  Terra, 
  default
</button>
`;

exports[`test should render a primary component 1`] = `
<button
  className="terra-Test terra-Test--primary"
  onClick={[Function]}>
  Terra, 
  primary
</button>
`;

exports[`test should toggle the class u-selected on default 1`] = `
<button
  className="terra-Test terra-Test--default"
  onClick={[Function]}>
  Terra, 
  default
</button>
`;

exports[`test should toggle the class u-selected on default 2`] = `
<button
  className="terra-Test terra-Test--default u-selected"
  onClick={[Function]}>
  Terra, 
  default
</button>
`;

exports[`test should toggle the class u-selected on default 3`] = `
<button
  className="terra-Test terra-Test--default"
  onClick={[Function]}>
  Terra, 
  default
</button>
`;

exports[`test should toggle the class u-selected on primary 1`] = `
<button
  className="terra-Test terra-Test--primary"
  onClick={[Function]}>
  Terra, 
  primary
</button>
`;

exports[`test should toggle the class u-selected on primary 2`] = `
<button
  className="terra-Test terra-Test--primary u-selected"
  onClick={[Function]}>
  Terra, 
  primary
</button>
`;

exports[`test should toggle the class u-selected on primary 3`] = `
<button
  className="terra-Test terra-Test--primary"
  onClick={[Function]}>
  Terra, 
  primary
</button>
`;
